src/test/java/org/minions/devfund/sergio/SquarelotronTest.java

```diff
@@ -0,0 +1,400 @@
+        final int size = 5;
+        squarelotron = new Squarelotron(size);
+        final int ringsExpected = 3;
+        assertEquals(squarelotron.getNumberOfRings(), ringsExpected);
+    }
+
+    /**
+     * Tests the values in the square of 3x3.
+     */
+    @Test
+    public void testRingCheckTrueInA3x3Square() {
+        final int size = 3;
+        squarelotron = new Squarelotron(size);
+        assertTrue(squarelotron.ringCheck(1, 1, 2));
+    }
+
+    /**
+     * Tests the values in the square of 4x4.
+     */
+    @Test
+    public void testRingCheckFalseInA4x4Square() {
+        final int size = 4;
+        squarelotron = new Squarelotron(size);
+        assertFalse(squarelotron.ringCheck(0, 0, 2));
+    }
+
+    /**
+     * Tests the values in the square of 5x5.
+     */
+    @Test
+    public void testUpsideDownFlipInA5x5SquareRingOne() {
+        final int size = 5;
+        squarelotron = new Squarelotron(size);
+        test.append("21 22 23 24 25").append("\n")
+                .append("16 07 08 09 20").append("\n")
+                .append("11 12 13 14 15").append("\n")
+                .append("06 17 18 19 10").append("\n")
+                .append("01 02 03 04 05");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.upsideDownFlip(1)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testUpsideDownFlipInA6x6SquareRingTwo() {
+        final int size = 6;
+        squarelotron = new Squarelotron(size);
+        test.append("01 02 03 04 05 06").append("\n")
+                .append("07 26 27 28 29 12").append("\n")
+                .append("13 20 15 16 23 18").append("\n")
+                .append("19 14 21 22 17 24").append("\n")
+                .append("25 08 09 10 11 30").append("\n")
+                .append("31 32 33 34 35 36");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.upsideDownFlip(2)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values in the square of 6x6.
+     */
+    @Test
+    public void testUpsideDownFlipIn6x6SquareRingThree() {
+        final int size = 6;
+        final int ring = 3;
+        squarelotron = new Squarelotron(size);
+        test.append("01 02 03 04 05 06").append("\n")
+                .append("07 08 09 10 11 12").append("\n")
+                .append("13 14 21 22 17 18").append("\n")
+                .append("19 20 15 16 23 24").append("\n")
+                .append("25 26 27 28 29 30").append("\n")
+                .append("31 32 33 34 35 36");
+        assertEquals(test.toString().trim(), squarelotron.showSquare(squarelotron.upsideDownFlip(ring)
+                .getOriginalSquareArray()));
+    }
+
+    /**
+     * Tests the values of the processed square of the method upsideDownFlip.
+     */
+    @Test
+    public void testUpsideDownFlipInOverCharge() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "testUpsideDownFlipInA5x5SquareRingOne" on line 93. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)